--- libxml2-2.7.6-32/configure.in.bak	Sat Feb  2 01:57:55 2013
+++ libxml2-2.7.6-32/configure.in	Sat Feb  2 01:58:18 2013
@@ -60,10 +60,6 @@
 AC_PATH_PROG(XMLLINT, xmllint, /usr/bin/xmllint)
 AC_PATH_PROG(XSLTPROC, xsltproc, /usr/bin/xsltproc)
 
-dnl Make sure we have an ANSI compiler
-AM_C_PROTOTYPES
-test "x$U" != "x" && AC_MSG_ERROR(Compiler not ANSI compliant)
-
 AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
 
